aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -147,6 +191,57 @@ void div_floor_kernel(TensorIteratorBase& iter) {
       });
     });
   } else {
+    // See NOTE: [Floor Division in Python]
+    if (iter.is_scalar(2) && at::isReducedFloatingType(dtype)) {
+      AT_DISPATCH_REDUCED_FLOATING_TYPES(dtype, "div_floor_cpu_reduced_float", [&]() {
+        float b = iter.original_scalar_value<float>(2);
+        iter.remove_operand(2);
+        using vec_t = Vectorized<float>;
+        cpu_kernel_vec(iter,
+          [=](scalar_t aa) __ubsan_ignore_float_divide_by_zero__ -> scalar_t {
```

Comment:
Can we share the logic instead of copying code from https://github.com/pytorch/pytorch/pull/98820/files#diff-0fd0c99904c62a1a7e46b08568d047c4e9ab153b529a8c6a2317ac6a3726bdffR249?

Reply:
updated~